Assets/MixedRealityToolkit.Services/InputSystem/Utilities/ScaleMeshEffect.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Microsoft.MixedReality.Toolkit.Input.Utilities
+{
+    /// <summary>
+    /// On Unity UI components the unity_ObjectToWorld matrix is not the transformation matrix of the local 
+    /// transform the Graphic component lives on, but that of it's parent Canvas. Many MRTK/Standard shader 
+    /// effects require object scale to be known. To solve this issue the ScaleMeshEffect will store scaling 
+    /// information into UV channel attributes during UI mesh construction. Ideally we would store the scale 
+    /// in one attribute but UGUI only supports two scalers per attribute (even in the tangent attribute).
+    /// </summary>
+    [RequireComponent(typeof(RectTransform))]
```

Comment:
Set helpurl? or better yet on BaseMeshEffect if not already there?

Reply:
Will do!